### PR TITLE
Make Flow modifications

### PR DESCRIPTION
**Pull request by drichmond**
* Removed a few unnecessary checks
* Moved rule declarations to more relevant files
* Added BSG_MACHINE_NAME to all Makefile.machine.include

@vb000 @tommydcjung - Can you give input on the new BSG_MACHINE_NAME variable? I'm using this to build machine-specific crt.o files so that someone can't run with the wrong crt.o file for a machine, among other things.

@mrutt92 for LGTM

### CODE REVIEW

File: Makefile.machine.include [hunk 1]
```diff
@@ -39,6 +39,9 @@ BSG_MACHINE_DRAM_BANK_SIZE_WORDS      = 134217728
 
 BSG_MACHINE_DATA_WIDTH                = 32
 
+# BSG Machine Name is used to build non-conflicting crt.o files
+BSG_MACHINE_NAME                      = Y$(BSG_MACHINE_GLOBAL_Y)X$(BSG_MACHINE_GLOBAL_X)_NONBLK_DDR4_F1MODEL
```
Comment (tommydcjung) [inline Makefile.machine.include]: I think it should be somewhat similar format as mem cfg names.
e.g.) y12_x8_vcache_non_blocking_ddr4_f1model

**Comment by tommydcjung** [comment]
why should the crt.o vary with different machine types?

**Comment by drichmond** [comment]
It was my impression that it linking against the wrong crt.o sometimes causes hangs: For example, Some machines have DRAM, some use Vcache. Some that have DRAM use different memory sizes.

File: Makefile.machine.include (hunk 1 above)
Comment (drichmond) [reply Makefile.machine.include]: I think this is good feedback. I'd be happy to do this too. 

I deferred to shorter names, but I'd be happy with your suggestion too.

**Comment by vb000** [comment]
crt has these implemented as symbols, afaik. Will take a look why these have to be different.

**Comment by drichmond** [comment]
It’s partly because of the .data location and stack pointer, right?

**Comment by vb000** [comment]
crt uses `_sp` symbol defined in the link script to initialize sp. So, should be independent of that...

**Comment by drichmond** [comment]
Hmmm. There’s no other arch specific part of crt.o? Is it just the link script?

Aside, can you rename link_dram2/dmem2 to be a little more descriptive?

**Comment by vb000** [comment]
No -- but I would rely on regression to prove it. Yes, it's just the link script that changes. Sure, I'll rename them.

File: Makefile.machine.include (hunk 1 above)
Comment (drichmond) [reply Makefile.machine.include]: I'm going to roll back the name addition. @vb000 is right, recompilation is not needed for crt.o

**Comment by drichmond** [comment]
Passes cosimulation on the default Makefile.machines.include and on 4x8_non_blocking_vcache_ramulator_hbm.

Also passes regression on F1

**Pull request merged**